src/main/java/org/takes/rq/CapInputStream.java

```diff
@@ -60,10 +59,7 @@
 
     @Override
     public int available() throws IOException {
-        return (int) Math.min(
-            (long) Integer.MAX_VALUE,
-            Math.max((long) this.origin.available(), this.more)
-        );
+        return (int) Math.min(this.origin.available(), this.more);
```

Comment:
@rui-castro even if this makes sense to me, this changes the whole initial logic of this method, so I let the architect decide. 
@yegor256 What do you think about this change?